### ISSUE TITLE
False Positive S1481 with RequestDelegate and HttpContext

### ISSUE THREAD

**Issue by JuanZamudioGBM** [SonarSource/sonar-dotnet#3295]
### Description

S1481 reported when a variable is used in a RequestDelegate.

### Repro steps
```
 using System;
using System.Threading.Tasks;
using Microsoft.AspNetCore.Http;
class Program
{
    private static void Main(string[] args)
    {
        var httpContext = new DefaultHttpContext(); // reported httpContext not used
        RequestDelegate requestDelegate = new RequestDelegate((httpContext) => { return Task.CompletedTask; });
        Console.WriteLine(requestDelegate.Target);
    }
}
```

### Expected behavior
Nothing reported
Please provide a description of the behavior you expect.

### Actual behavior
s1481 reported

### Known workarounds
None

### Related information
* C#/VB.NET Plugins Version 8.6.0.16497 aanalyzer
* VS 16.5.2
* dotnet 3.1
* Windows 10

**Comment by pavel-mikula-sonarsource** [comment]
Hi @JuanZamudioGBM ,

Problem with your code is that you're using duplicate variable name in the same scope. You declare `httpContext` variable on first line and then declare independent `httpContext` lambda parameter on second line. So your first variable is really not used and rule behaves as expected here. The code will compile even if you comment the variable declaration.
